pkg/snapshotmgr/snapshot_manager.go

```diff
@@ -168,11 +176,11 @@ func NewSnapshotManagerFromConfig(configInfo server.ConfigInfo, s3RepoParams map
 		dpem.RegisterExternalProtectedEntityTypeManagers([]astrolabe.ProtectedEntityTypeManager{paraVirtPETM})
 	}
 
-	snapMgr := SnapshotManager{
-		FieldLogger: logger,
-		config:      config,
-		Pem:         dpem,
-		s3PETM:      s3PETM,
+	ivdPETM := dpem.GetProtectedEntityTypeManager("ivd")
+	if ivdPETM != nil {
+		dmIVDPE := NewDataManagerProtectedEntityTypeManager(ivdPETM, &snapMgr, true)
+		// We use a DataManagerPE to delegate data copy to the data manager
+		dpem.RegisterExternalProtectedEntityTypeManagers([]astrolabe.ProtectedEntityTypeManager{dmIVDPE})
```

Comment:
@lintongj a new external pe type manager is being registered here. Could you check if https://github.com/vmware-tanzu/astrolabe/blob/master/pkg/pvc/pvc_protected_entity.go#L360 breaks?

Reply:
Yes. It would be problematic in guest cluster. Anyway, the getComponentPEType function itself (at https://github.com/vmware-tanzu/astrolabe/blob/master/pkg/pvc/pvc_protected_entity.go#L360) is hacky. But, I didn't think of any better solution without hardcoding since ParaVirt PETM is externally registered. 

In this version of code, ivdPETM will be initialized in any flavors of cluster. ivdPETM will not be used in guest cluster. Do we really need to do that?

Reply:
We should change it to register as part of the setup.  As it is, the dmIVDPETM overrides the ivd PETM for type ivd so it doesn't actually change anything.  The original ivd PETM is no longer visible.

Reply:
I misread the line L179 above,
```
ivdPETM := dpem.GetProtectedEntityTypeManager("ivd")
```
I somehow though it would initialize an IVD PETM. Now, it looks good to me.